Index: MCBBS CreditAnalysis.user.js
===================================================================
--- MCBBS CreditAnalysis.user.js
+++ MCBBS CreditAnalysis.user.js
@@ -9,9 +9,16 @@
 // @match        https://*.mcbbs.net/home.php?mod=space&username=*
 // @grant        none
 // ==/UserScript==
 
+
 (function() {
+    // CaveNightingale - begin
+    const creditRegex = /<li><em>积分<\/em>(-?[0-9]+)<\/li><li><em>人气<\/em>(-?[0-9]+) 点<\/li>\n<li><em>金粒<\/em>(-?[0-9]+) 粒<\/li>\n<li><em>金锭\[已弃用\]<\/em>(-?[0-9]+) 块<\/li>\n<li><em>宝石<\/em>(-?[0-9]+) 颗<\/li>\n<li><em>下界之星<\/em>(-?[0-9]+) 枚<\/li>\n<li><em>贡献<\/em>(-?[0-9]+) 份<\/li>\n<li><em>爱心<\/em>(-?[0-9]+) 心<\/li>\n<li><em>钻石<\/em>(-?[0-9]+) 颗<\/li>/;
+    const creditList = ['积分', '人气', '金粒', '金锭', '宝石', '下界之心', '贡献', '爱心', '钻石'];
+    const postsRegex = / target="_blank">回帖数 (-?[0-9]+?)<\/a>/;
+    const threadsRegex = / target="_blank">主题数 (-?[0-9]+?)<\/a>/;
+    // CaveNightingale - end
 
     console.log(" %c MCBBS %c CreditAnalysis", "color: #fff; background: #f8981d; padding:5px;", "color:#fff; background: #000; padding:5px;");
     console.log(" %c Made by %c 快乐小方 ", "color: #fff; background: #815098; padding:5px;", "color:#fff; background: #000; padding:5px;");
     console.group('MCA Log');
@@ -23,21 +30,39 @@
     console.log("[L] 用户UID：" + uid);
 
     console.log("[L] 开始获取用户数据");
     // 调用api
-    jq.ajax({
-        type:'get',
-        url:"https://www.mcbbs.net/api/mobile/index.php?module=profile&uid=" + uid,
-        success:function(body,heads,status){
+    // CaveNightingale - begin
+    // copyed form netlify/functions/get-mcbbs-credit.js
+    console.log("[L] 正在运行的CreditAnalysis是洞穴夜莺补丁版本");
+    let data = document.body.innerHTML;
+    let result = {};
+    let creditParsed = creditRegex.exec(data);
+    if (creditParsed) {
+        let credits = {};
+        for (let i = 0; i < creditList.length; i++)
+            credits[creditList[i]] = parseInt(creditParsed[i + 1]);
+        result.credits = credits;
+    }
+    let postsParsed = postsRegex.exec(data);
+    if (postsParsed)
+        result.posts = parseInt(postsParsed[1]);
+    let threadsParsed = threadsRegex.exec(data);
+    if (threadsParsed)
+        result.threads = parseInt(threadsParsed[1]);
+    // CaveNightingale - end
             console.log("[L] 成功获取用户数据");
-            var credits = body.Variables.space.credits;            //积分总值
-            var extcredits1 = body.Variables.space.extcredits1;    //人气
-            var extcredits2 = body.Variables.space.extcredits6;    //贡献
-            var extcredits3 = body.Variables.space.extcredits7;    //爱心
-            var extcredits4 = body.Variables.space.extcredits8;    //钻石
-            var posts = body.Variables.space.posts;                //发帖数
-            var threads = body.Variables.space.threads;            //主题数
-            var digestposts = body.Variables.space.digestposts;    //精华数
+            // CaveNightingale - begin
+            var credits = result.credits['积分'];            //积分总值
+            var extcredits1 = result.credits['人气'];    //人气
+            var extcredits2 = result.credits['贡献'];    //贡献
+            var extcredits3 = result.credits['爱心'];    //爱心
+            var extcredits4 = result.credits['钻石'];    //钻石
+            var posts = result.posts + result.threads;                //发帖数
+            var threads = result.threads;            //主题数
+            var tempSum = Math.round(extcredits1*3 + extcredits2*10 + extcredits3*4 + extcredits4*2 + posts / 3 + threads * 2);
+            var digestposts = (credits - tempSum) / 45;    //精华数
+            // CaveNightingale - end
 
             //DeBug：console.log("[D] 人气分:" + extcredits1*3 + "\n贡献分:" + extcredits2*10 + "\n爱心分:" + extcredits3*4 + "\n钻石分:" + extcredits4*2 + "\n发帖分:" + Math.round(posts/3) + "\n主题分:" + threads*2 + "\n精华分:" + digestposts*45)
             var json = [credits,extcredits1*3,extcredits2*10,extcredits3*4,extcredits4*2,posts/3,threads*2,digestposts*45];
             console.log("[D] 用户积分数：" + json[0] +"，公式计算得：" + Math.round(json[1] + json[2] + json[3] + json[4] + json[5] + json[6] + json[7]));
@@ -276,8 +301,5 @@
 	console.log("[L] 绘制完成");
 
 </script>
 `);
-            console.groupEnd();
-        }
-    });
 })();
\ No newline at end of file
